### PR TITLE
Qemu cleanup and tests

### PR DESCRIPTION
**Pull request by SwampDragons**
Qemu was overdue for a little cleanup. 

This pr: 
- Removes config and config tests into their own files, as is common in the other builders
- Adds a few tests to step_run, including tests of recently added cd_files logic
- Removes most of the conditional logic from the builder's multistep construction, and pushes that logic into the steps themselves. This makes it easier to see the shape of the build, and allows us to add tests for those conditionals into the more-easily-tested steps.

### CODE REVIEW

**Review by nywilken — APPROVED** [review]
This looks great! Good job on breaking out the config, and moving the conditionals into each of the individual steps. 👍

**Comment by nywilken** [comment]
Also 100000+ on the added tests.

**Review by azr — APPROVED** [review]
Love it ! Nice adding tests.I was reading the commit history which made it really nice to review.

File: builder/qemu/step_run_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	state.Put("vnc_port", 5905)
+	state.Put("vnc_password", "fake_vnc_password")
+
+	return state
+}
+
+func Test_getCommandArgs(t *testing.T) {
+	state := runTestState(t, &Config{})
+
+	args, err := getCommandArgs("", state)
```
Comment (azr) [inline builder/qemu/step_run_test.go:37]: Nice, getCommandArgs has sooo many cases; it's great to test it 👍

File: builder/qemu/builder.go [hunk 2]
```diff
@@ -595,15 +41,6 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
...
-	steprun := &stepRun{}
-	if !b.config.DiskImage {
-		steprun.BootDrive = "once=d"
-		steprun.Message = "Starting VM, booting from CD-ROM"
-	} else {
-		steprun.BootDrive = "c"
-		steprun.Message = "Starting VM, booting disk image"
-	}
-
 	steps := []multistep.Step{}
```
Comment (sylviamoss) [inline builder/qemu/builder.go:44]: So clean now 😍

**Review by sylviamoss — APPROVED** [review]
Super nice!! I will be much easier to work from now on 👍

File: builder/qemu/step_run_test.go (hunk 1 above)
Comment (SwampDragons) [reply builder/qemu/step_run_test.go:37]: This doesn't even begin to cover them all but it's a start and at least covers the cd_files option I added last week

**Pull request merged**